CMakeLists.txt

```diff
@@ -639,6 +640,49 @@ PUBLIC
 
 add_subdirectory(interface/basisu_c_binding)
 
+# Only one architecture is supported at once, if neither of
+# ISA_SSE41 and ISA_SSE2  are defined ISA_AVX2 is chosen.
+# If ISA_AVX2 fails to compile user must chose other x86 options.
+# On arm based systems ISA_NEON is default
+
+list(FIND CMAKE_OSX_ARCHITECTURES "$(ARCHS_STANDARD)" ASTC_BUILD_UNIVERSAL)
+
+if(${ASTC_BUILD_UNIVERSAL} EQUAL -1)
+    if (${ISA_NONE})
```

Comment:
If ISA_NONE etc are not defined in the cmake configure command, what is the default value and where is it set? Are these CMake cache variables?

Reply:
These are cache variables defined [here](https://github.com/KhronosGroup/KTX-Software/blob/master/lib/astc-encoder/CMakeLists.txt#L41). Its value is `OFF` by default. Its a way for clients to enforce they don't want any SIMD options. In this context its used for detection the encoder shared library name. `set(ASTC_LIB_NAME astcenc-none-static)`. This is another quirk of the encoder build system and ability to build multiple binaries from one configure.

Reply:
Is asct-encoder/CMakeLists.txt used in the KTX-Software build?

Reply:
It is used by add_subdirectory [here](https://github.com/wasimabbas-arm/KTX-Software/blob/astc_support/CMakeLists.txt#L682)

Reply:
Shouldn't this add_subdirectory be before line 650? That's where various tests on the ISA values start?

Reply:
No because once `astc-encoder/CMakeLists.txt` is invovked, it expects all the various ISA options to be set properly.

Reply:
I thought the `option` declarations in `lib/astc-encoder/CMakeLists.txt` were setting default values. Now I see they aren't so the order of inclusion isn't currently important.

Now If nothing is specified to `cmake` and the CPU is Intel x64 the code in `CMakeLists.txt` will set `ISA_AVX2` `ON`. Is AVX2 more widespread than the SSE options? Is it faster? How are users supposed to know whether they should choose `ISA_SSE2` or `ISA_SSE41` instead?

Reply:
There is some information about this in the ASTC encoder readme. 

```
For x86-64 we provide, in order of increasing performance:

    `astcenc-sse2` - uses SSE2
    `astcenc-sse4.1` - uses SSE4.1 and POPCNT
    `astcenc-avx2` - uses AVX2, SSE4.2, POPCNT, and F16C

The x86-64 SSE2 builds will work on all x86-64 machines, but it is the slowest of the three. The other two require extended CPU instruction set support which is not universally available, but each step gains ~15% more performance.

For Apple silicon macOS devices we provide:

    `astcenc-neon` - uses NEON
```

We could replicate that in ktx-software readme if we want to keep the same options. But I think we should clean these options up. at the very least hid these behind generic KTX-Software specific cmake options.

Reply:
> But I think we should clean these options up. at the very least hid these behind generic KTX-Software specific cmake options.

I agree. Let's figure out the universal binary build, see what effect it has on the options and then we can make some generic KTX-Software cmake options.